### PR TITLE
DOC: Add N-dimensional argmax/argmin example

### PR DESCRIPTION
**Pull request by pmvz**
Implements #23162. I have built the docs with Gitpod and found no problems.

### CODE REVIEW

**Review by Mukulikaa — COMMENTED** [review]
Thanks for the PR, @pmvz! We try to keep less than 75 characters per line so it would be great if you could reformat that. I also left a few stylistic suggestions.  Apart from that, LGTM.

File: doc/source/user/how-to-index.rst
Comment (pmvz) [reply doc/source/user/how-to-index.rst]: I considered this but opted not to because they were already linked in the same section. For future reference, is it always necessary to link to methods that have already been linked to within a few lines?

Comment (Mukulikaa) [reply doc/source/user/how-to-index.rst]: It is always recommended because it increases accessibility on smaller screens where you have to scroll quite a bit even within a section.

**Review by Mukulikaa — APPROVED** [review]
LGTM. I'll let @rossbar or @seberg take a look before merging!

Edit: The failing test seems unrelated.

**Comment by mattip** [comment]
Thanks @pmvz

**Comment by InessaPawson** [comment]
Hi-five on merging your first pull request to NumPy, @pmvz! We hope you stick around! Your choices aren’t limited to writing code – you can review pull requests, help us stay on top of new and old  issues, develop educational material, work on our website, add or improve graphic design, create marketing materials, translate website content, write grant proposals, and help with other fundraising initiatives. For more info, check out: https://numpy.org/contribute.
Also, consider joining our mailing list. This is a great way to connect with other cool people in our community and be part of important conversations that affect the development of NumPy: https://mail.python.org/mailman/listinfo/numpy-discussion.

**Pull request merged**
